Do not mention automatic update when only dealing with update check (#18118)
nvaccess/nvda: user_docs/en/userGuide.md

| Option | Behaviour |
|---|---|
| Same | Add-ons will remain on their channel |
| Any | Add-ons will check for update to the latest version, regardless of channel |
| Do not update | Add-ons will not check for update by default, you must enable them individually |
| Stable | Add-ons will check for update to stable versions |
| Beta or dev | Add-ons will check for update to beta or dev versions |
| Beta | Add-ons will check for update to beta versions |
| Dev | Add-ons will check for update to dev versions |

##### Allow incompatible add-ons when checking for updates {#AllowIncompatibleAddonUpdates}

> When NVDA checks for add-on updates, this setting allows to accept updates to add-ons that may not be fully compatible with the current version of NVDA.

Reviewer: ```suggestion
-This setting allows NVDA to update add-ons to versions that may not be fully compatible with the current version of NVDA when checking for add-on updates.
```

By default, this is disabled, meaning add-ons update will only check for add-on versions marked as compatible with the current version of NVDA.

> Add-on updates check will still allow to update an incompatible add-on version to a compatible version when it is released.

Reviewer: ```suggestion
Add-on update checks will still include updates from incompatible add-on versions to compatible versions when they are released.
```
